proposals/generate_validating_admission_policies.md

```diff
@@ -468,9 +473,171 @@ spec:
+
+Excluding resources in ValidatingAdmissionPolicy can be done via `spec.matchConstraints.excludeResourceRules`.
+
+Here is an example of ValidatingAdmissionPolicy which excludes deployments named `staging`:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "check-deployment-labels"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["apps"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["deployments"]
+    excludeResourceRules:
+    - apiGroups:     ["apps"]
+      apiVersions:   ["v1"]
+      operations:    ["CREATE", "UPDATE"]
+      resources:     ["deployments"]
+      resourceNames: ["staging"]
+  validations:
+    - expression: "'app' in object.metadata.labels"
+```
+
+Lets try generating a ValidatingAdmissionPolicy from a PolicyException.
+
+- Given an exception that excludes Deployments named `important-tool` in a namespace whose `environment` label is set to `staging`.
+
+  ```yaml
+  apiVersion: kyverno.io/v2beta1
+  kind: PolicyException
+  metadata:
+    name: delta-exception
+    namespace: delta
+  spec:
+    exceptions:
+    - policyName: disallow-host-namespaces
+      ruleNames:
+      - host-namespaces
+    match:
+      any:
+      - resources:
+          kinds:
+          - Deployment
+          names:
+          - important-tool
+          namespaceSelector:
+            matchLabels:
+              environment: staging
+  ```
+
+  The VAP has the ability to select/exclude resources via the namespaceSelector as follows:
+
+  ```yaml
+  apiVersion: admissionregistration.k8s.io/v1beta1
+  kind: ValidatingAdmissionPolicy
+  metadata:
+    name: "check-deployment-labels"
+  spec:
+    failurePolicy: Fail
+    matchConstraints:
+      resourceRules:
+      - apiGroups:   ["apps"]
+        apiVersions: ["v1"]
+        operations:  ["CREATE", "UPDATE"]
+        resources:   ["deployments"]
+      excludeResourceRules:
+      - apiGroups:     ["apps"]
+        apiVersions:   ["v1"]
+        operations:    ["CREATE", "UPDATE"]
+        resources:     ["deployments"]
+        resourceNames: ["important-tool"]
+      namespaceSelector:
+        matchLabels:
+          environment: staging
+    validations:
+      - expression: "'app' in object.metadata.labels"
+  ```
+
+  However, using a `namespaceSelector` will be applied for both matching and excluding deployemnts. The above VAP matches all deployments whose namespace labels set to `environment: staging`, and it also excludes all deployments named `important-tool` whose namespace labels set to `environment: staging`. It means that matching/excluding resources is done in the same namespace. There is no option to match deployments in all namespaces and exclude some in a specific namespace.
```

Comment:
@MariamFahmy98 - isn't this the same as what this Kyverno declaration would do?

```yaml
...
  rules:
  - name: check-deployment-labels
    match:
      all:
      - resources:
          namespaceSelector:
            matchLabels:
              environments: staging
    exclude:
      all:
        - resources:
            name: important-tool
```

Reply:
The corresponding VAP could be something like this:

Reply:
The corresponding VAP could be something like this:
```
apiVersion: admissionregistration.k8s.io/v1beta1
kind: ValidatingAdmissionPolicy
metadata:
  name: "check-deployment-labels"
spec:
  failurePolicy: Fail
  matchConstraints:
    excludeResourceRules:
    - apiGroups:     ['*']
      apiVersions:   ['*']
      resources:     ['*']
      resourceNames: ["important-tool"]
    namespaceSelector:
      matchLabels:
        environment: staging
  validations:
    - expression: "'app' in object.metadata.labels"
```
But I need to double-check if this is correct.